# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/compiler/translator/CollectVariables.cpp
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/compiler/translator/CollectVariables.cpp
--- a/gfx/angle/checkout/src/compiler/translator/CollectVariables.cpp	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/compiler/translator/CollectVariables.cpp	Fri Nov 08 17:06:00 2019 +0000
@@ -175,6 +175,8 @@
     bool mVertexIDAdded;
     bool mPointSizeAdded;
     bool mDrawIDAdded;
+    bool mBaseVertexAdded;
+    bool mBaseInstanceAdded;
 
     // Vertex Shader and Geometry Shader builtins
     bool mPositionAdded;
@@ -233,6 +235,8 @@
       mVertexIDAdded(false),
       mPointSizeAdded(false),
       mDrawIDAdded(false),
+      mBaseVertexAdded(false),
+      mBaseInstanceAdded(false),
       mPositionAdded(false),
       mPointCoordAdded(false),
       mFrontFacingAdded(false),
@@ -489,6 +493,12 @@
             case EvqDrawID:
                 recordBuiltInAttributeUsed(symbol->variable(), &mDrawIDAdded);
                 return;
+            case EvqBaseVertex:
+                recordBuiltInAttributeUsed(symbol->variable(), &mBaseVertexAdded);
+                return;
+            case EvqBaseInstance:
+                recordBuiltInAttributeUsed(symbol->variable(), &mBaseInstanceAdded);
+                return;
             case EvqLastFragData:
                 recordBuiltInVaryingUsed(symbol->variable(), &mLastFragDataAdded, mInputVaryings);
                 return;
@@ -554,7 +564,8 @@
                 else
                 {
                     ASSERT(mShaderType == GL_VERTEX_SHADER &&
-                           IsExtensionEnabled(mExtensionBehavior, TExtension::OVR_multiview2));
+                           (IsExtensionEnabled(mExtensionBehavior, TExtension::OVR_multiview2) ||
+                            IsExtensionEnabled(mExtensionBehavior, TExtension::OVR_multiview)));
                 }
                 break;
             default: